kube-runtime/src/reflector/store.rs

```diff
@@ -127,8 +127,26 @@ where
         let s = self.store.read();
         s.values().cloned().collect()
     }
-}
 
+    /// Retrieve a `clone()` of the entry found by the given predicate
+    #[must_use]
+    pub fn find<P>(&self, predicate: P) -> Option<Arc<K>>
+    where
+        P: Fn(&K) -> bool,
+    {
+        self.store
+            .read()
+            .iter()
+            .map(|(_, k)| k)
+            .find(|k| predicate(k.as_ref()))
+            .cloned()
+    }
```

Comment:
i like this signature. It is a pretty big improvement if you don't need a full clone, however it does need to take the read lock, but that's a reasonable tradeoff.